src/v/cluster/controller.cc

```diff
@@ -96,6 +98,8 @@ controller::controller(
   , _node_status_table(node_status_table)
   , _probe(*this) {}
 
+controller::~controller() {}
```

Comment:
Why is this explicit destructor needed?

Reply:
Will add a comment, but the controller destructor needs to be able to destruct the `cloud_metadata::uploader`, which I didn't want to include as a header here. I guess it doesn't matter too much though, given they're all in v::cluster right now anyway